src/ttvchat/TwitchChatService.ts

```diff
@@ -0,0 +1,128 @@
+import * as vscode from 'vscode';
+
+import { HighlighterAPI } from '../index';
+import { AuthenticationService } from './AuthenticationService';
+import { Logger, log } from '../logger';
+import { Commands, Configuration, Settings } from '../enums';
+import { ChatClient, ChatClientMessageReceivedEvent } from './ChatClient';
+import { parseMessage } from '../utils';
+
+export class TwitchChatService implements vscode.Disposable {
+  private readonly _api: HighlighterAPI;
+  private readonly _authenticationService: AuthenticationService;
+
+  private log: log;
+  private loginStatusBarItem: vscode.StatusBarItem;
+  private chatClientStatusBarItem: vscode.StatusBarItem;
+  private chatClient: ChatClient;
+  private config?: vscode.WorkspaceConfiguration;
+
+  constructor(api: HighlighterAPI, outputChannel: vscode.OutputChannel) {
+    this.log = new Logger(outputChannel).log;
+    this.chatClient = new ChatClient(this.log);
+    this.chatClient.disconnect.bind(this.chatClient);
```

Comment:
I forget what does the `.bind` do here?

Reply:
The .bind sets the context of the _this_ keyword. In this case, we don't want the _this_ keyword to be in the context of the _TwitchChatService_ object but rather the _ChatClient_ object.